daal4py/sklearn/linear_model/logistic_path.py

```diff
@@ -596,15 +594,32 @@ def daal4py_predict(self, X, resultsToEvaluate):
     except ValueError:
         fptype = None
 
-    daal_ready = self.multi_class in ["multinomial", "warn"] or \
-        self.classes_.size == 2 or resultsToEvaluate == 'computeClassLabels'
-    daal_ready = daal_ready and not sparse.issparse(X) and \
-        not sparse.issparse(self.coef_) and fptype is not None
-    if daal_ready:
-        logging.info(
-            "sklearn.linear_model.LogisticRegression."
-            "predict: " + get_patch_message("daal")
-        )
+    if resultsToEvaluate == 'computeClassLabels':
+        _function_name = 'predict'
+    elif resultsToEvaluate == 'computeClassProbabilities':
+        _function_name = 'predict_proba'
+    elif resultsToEvaluate == 'computeClassLogProbabilities':
+        _function_name = 'predict_log_proba'
+    else:
+        raise ValueError('resultsToEvaluate must be in [computeClassLabels, \
+            computeClassProbabilities, computeClassLogProbabilities]')
+
+    _patching_status = PatchingConditionsChain(
+        f"sklearn.linear_model.LogisticRegression.{_function_name}")
+    _patching_status.and_conditions([
+        (self.multi_class in ["multinomial", "warn"],
+            "multiclass is not 'multinomial' or 'warn'"),
+        (self.classes_.size == 2, "self.classes_.size != 2"),
+        (resultsToEvaluate == 'computeClassLabels',
+            "resultsToEvaluate != 'computeClassLabels'")],
+        conditions_merging=any)
+    _dal_ready = _patching_status.and_conditions([
+        (not sparse.issparse(X), "input is sparse"),
+        (not sparse.issparse(self.coef_), "self.coef_ is sparse"),
+        (fptype is not None, "fptype is incorrect")])
```

Comment:
It might be better to use full phrase, the suggestion is to use **floating point type is incorrect**, please fix in all places

Reply:
I moved to two types of messages:
 - where `fptype = getFPType(X)` is used: `unable to get dtype`
 - where `check_array(X).dtype` is used: `X data type is {X.dtype} while np.float32 and np.float64 are supported`